File: Compiler/src/main/kotlin/nodes/StmtSeqNode.java
package nodes;

public class StmtSeqNode {
    public StmtNode first;
    public StmtNode last;

    //Для Пустого elseIf и пустого stmt_seq
    public StmtSeqNode() {
        this.first = null;
        this.last = null;
    }

    public void addStmt(StmtNode stmt) {
        if(this.last == null) {
            this.first = stmt;
            this.last = stmt;
        }
        else {
            this.last.next = stmt;
            this.last = stmt;
        }
    }

    public void addElseifSeq(ExprNode expr, StmtSeqNode block) {
        StmtNode else_if = new StmtNode();
        else_if.id = Utils.getLastId();
        else_if.condition_expr = expr;
        else_if.if_block = block;

        if(this.last == null) {
            this.first = else_if;
            this.last = else_if;
        }
        else {
            this.last.next = else_if;
            this.last = else_if;
        }
    }
}

File: Compiler/src/main/kotlin/nodes/StmtSeqNode.kt
package nodes

class StmtSeqNode(val startID: Int, var lastID: Int)    //Для Пустого elseIf и пустого stmt_seq
{
    var first: StmtNode? = null
    var last: StmtNode? = null
    fun addStmt(stmt: StmtNode?) {
        if (last == null) {
            first = stmt
            last = stmt
        } else {
            last!!.next = stmt
            last = stmt
        }
    }

    fun addElseifSeq(expr: ExprNode?, block: StmtSeqNode?) {
        val elseIf = StmtNode()
        elseIf.id = Utils.lastId
        elseIf.conditionExpr = expr
        elseIf.ifBlock = block
        if (last == null) {
            first = elseIf
            last = elseIf
        } else {
            last!!.next = elseIf
            last = elseIf
        }
    }
}